### PR TITLE
DIV-4398 - Remove documents from original case before copying case data into draft

### PR DESCRIPTION
**Pull request by jvictorcf**
# Description

When amending a petition, we can't copy any documents from the original case.

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Unit tests.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java [hunk 1]
```diff
@@ -1,141 +1,143 @@
...
-    IssueDate("IssueDate"),
-    PermittedDecreeNisiReason("PermittedDecreeNisiReason"),
-    DNApplyForDecreeNisi("DNApplyForDecreeNisi"),
-    PreviousReasonsForDivorce("PreviousReasonsForDivorce"),
-    D8DocumentsGenerated("D8DocumentsGenerated"),
-    D8SeparationReferenceDate("D8SeparationReferenceDate");
+/*
+ * These values are specifically for Amend Petition workflow, NOT for general usage
+ */
+public class AmendCaseRemovedProps {
```
Comment (alexwakeman) [inline src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java]: Could use `@NoArgsConstructor` ?

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java [hunk 2]
```diff
@@ -1,141 +1,143 @@
...
+        "PreviousReasonsForDivorce",
+        "D8DocumentsGenerated",
+        "D8DocumentsUploaded",
+        "D8RejectDocumentsUploaded",
+        "D8SeparationReferenceDate"
+    };
 
-    public static Stream<String> getProps() {
-        return Arrays.stream(AmendCaseRemovedProps.values()).map(AmendCaseRemovedProps::getValue);
+    public static Stream<String> getPropertiesToRemoveAsStream() {
```
Comment (gurpiarbassi) [inline src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java]: bad practice to return a stream from a public method. but 😞  it's been like that for a long time

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java (hunk 1 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java]: Wouldn't that implement the public constructor, @alexwakeman? Sonar complained that I should implement the private no-args constructor.

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java (hunk 2 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java]: That's easy enough to change. I'll do it.

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java (hunk 1 above)
Comment (alexwakeman) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/AmendCaseRemovedProps.java]: Yes could well be fair enough 👍

**Pull request merged**